lua/autorun/server/sv_restrict_propcore.lua

```diff
@@ -1,41 +1,40 @@
--- Propcore is allowed to everyone, but functions in the restrictedFunctions array will be restricted to devotee+ only
+-- Restricts PropCore functions based on rank
 
+local restrictedFunctions, rankRestrictedFunctions = require "sv_restricted_propcore_functions"
 local function restrictPropCoreFunctions()
-    local disallowedRanks = {}
-    disallowedRanks["user"] = true
-    disallowedRanks["regular"] = true
-
-    local restrictedFunctions = {
-        "propSpawn(sn)",
-        "propSpawn(en)",
-        "propSpawn(svn)",
-        "propSpawn(evn)",
-        "propSpawn(san)",
-        "propSpawn(ean)",
-        "propSpawn(svan)",
-        "propSpawn(evan)",
-        "seatSpawn(sn)",
-        "seatSpawn(svan)",
-        "setPos(e:v)",
-        "reposition(e:v)",
-        "propManipulate(e:vannn)"
-        --"propBreak(e:)"
-    }
-
-    for _, signature in pairs( restrictedFunctions ) do
+    for _, signature in pairs( RESTRICTED_FUNCTIONS ) do
         local oldFunc = wire_expression2_funcs[signature][3]
-
         wire_expression2_funcs[signature][3] = function( self, ... )
+            local playerGroup       = self.player:GetUserGroup()
+            local groupRestrictions = rankRestrictedFunctions[playerGroup] or rankRestrictedFunctions['user']
+            local restrictions      = groupRestrictions[signature]
+
             if ( disallowedRanks[self.player:GetUserGroup()] == nil ) then
```

Comment:
disallowedRanks ranks isn't defined  and this statement seems to have no affect as isInBuildMode is defined later in the function